src/Stratis.Bitcoin.Features.Consensus/PowConsensusValidator.cs

```diff
@@ -287,6 +313,11 @@ public virtual void ExecuteBlock(ContextInformation context, TaskScheduler taskS
             this.logger.LogTrace("(-)");
         }
 
+        /// <summary>
+        /// Updates context's coin view.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="tx">Transaction that is used for fetching UTXO's.</param>
```

Comment:
i don't think this is correct
seems like the transaction here is the one which inputs are to be updated
check the code how it is used

Reply:
This is called from ExecuteBlock for each block's transaction. And what it does is it just adds or replaces transaction in context's UnspentOutputSet. 
I don't see here any code where transaction's inputs get updated.

Reply:
I see, misunderstanding here. I did not mean to say that the transaction itself is being updated.
CoinView is being updated. What this method does is that we have previous CV state C1 and then we have transaction X, and the result is CV state C2 as in `f(C1, X) -> C2`

state C2 is exactly the same as C1 except for those outputs that are being spent using X (X's inputs), so those outputs are no longer in C2 and X's outputs are new to C2 and are not in C1

so `C2 := C1 - {X_inputs} + {X_outputs}`

Reply:
and again which context?